core/app/presenters/spree/filters/options_presenter.rb

```diff
@@ -9,7 +9,7 @@ def to_h
             id: option_type.id,
             name: option_type.name,
             presentation: option_type.presentation,
-            option_values: option_values.map { |e| option_value_hash(e) }
+            option_values: option_values.map { |e| option_value_hash(e) }.sort_by { |e| e[:position] }
```

Comment:
@aleksandarpetrushev rather than doing this here - please fix it at the source - https://github.com/spree/spree/blob/a0137d7aa25fa48639c88e5baef11bd236559812/core/app/finders/concerns/spree/product_filterable.rb#L6